optuna/importance/_fanova/_fanova.py

```diff
@@ -0,0 +1,262 @@
+        min_samples_leaf: Union[int, float],
+        random_state: Optional[int],
+    ) -> None:
+        _check_sklearn_availability()
+
+        self._forest = RandomForestRegressor(
+            n_estimators=n_estimators,
+            max_depth=max_depth,
+            min_samples_split=min_samples_split,
+            min_samples_leaf=min_samples_leaf,
+            random_state=random_state,
+        )
+        self._trees = None  # type: Optional[List[_FanovaTree]]
+        self._V_U_total = None  # type: Optional[Dict[Tuple[int, ...], numpy.ndarray]]
+        self._V_U_individual = None  # type: Optional[Dict[Tuple[int, ...], numpy.ndarray]]
+        self._features_to_raw_features = None  # type: Optional[List[numpy.ndarray]]
+
+    def fit(
+        self,
+        X: numpy.ndarray,
+        y: numpy.ndarray,
+        search_spaces: numpy.ndarray,
+        search_spaces_is_categorical: List[bool],
+    ) -> None:
+        assert X.shape[0] == y.shape[0]
+        assert X.shape[1] == search_spaces.shape[0]
+        assert X.shape[1] == len(search_spaces_is_categorical)
+        assert search_spaces.shape[1] == 2
+
+        encoder = _CategoricalFeaturesOneHotEncoder()
+        X, search_spaces = encoder.fit_transform(X, search_spaces, search_spaces_is_categorical)
+
+        self._forest.fit(X, y)
+
+        self._trees = [_FanovaTree(e.tree_, search_spaces) for e in self._forest.estimators_]
+        self._V_U_total = {}
+        self._V_U_individual = {}
+        self._features_to_raw_features = encoder.features_to_raw_features
+
+    def quantify_importance(
+        self, features: Tuple[int, ...]
+    ) -> Dict[Tuple[int, ...], Dict[str, float]]:
+        assert self._trees is not None
+        assert self._V_U_total is not None
+        assert self._V_U_individual is not None
+
+        if all(tree.variance == 0 for tree in self._trees):
+            # If all trees have 0 variance, we cannot assess any importances.
+            # This could occur if for instance `X.shape[0] == 1`.
+            raise RuntimeError("Encountered zero total variance in all trees.")
+
+        self._compute_marginals(features)
+
+        importance_dict = {}  # type: Dict[Tuple[int, ...], Dict[str, float]]
+
+        for k in range(1, len(features) + 1):
+            for sub_features in itertools.combinations(features, k):
+                importance_dict[sub_features] = {}
+
+                fractions_total = []
+                fractions_individual = []
+                for tree_index, tree in enumerate(self._trees):
+                    tree_variance = tree.variance
+                    if tree_variance == 0:
+                        continue
+                    fractions_total.append(
+                        self._V_U_total[sub_features][tree_index] / tree_variance
+                    )
+                    fractions_individual.append(
+                        self._V_U_individual[sub_features][tree_index] / tree_variance
+                    )
```

Comment:
Would you like to switch the order of `fractions_total` and `fractions_individual`? The other orders are `fractions_individual` -> `fractions_total`.
```suggestion
                    fractions_individual.append(
                        self._V_U_individual[sub_features][tree_index] / tree_variance
                    )
                    fractions_total.append(
                        self._V_U_total[sub_features][tree_index] / tree_variance
                    )
```

Reply:
Makes sense. I added your suggestion.